pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1082,7 +1083,19 @@ public void checkGC(int gcIntervalInSeconds) {
         producers.forEach(Producer::checkPermissions);
         subscriptions.forEach((subName, sub) -> sub.getConsumers().forEach(Consumer::checkPermissions));
         checkMessageExpiry();
-        return checkReplication();
+        CompletableFuture<Void> result = new CompletableFuture<Void>();
+        checkReplication().thenAccept(res -> {
+            log.info("Policies updated successfully {}", data);
+            result.complete(null);
+        }).exceptionally(th -> {
+            log.error("Policies update failed {} {}, scheduled retry in {} seconds", data, th.getMessage(),
+                    POLICY_UPDATE_FAILURE_RETRY_TIME_SECONDS, th);
+            brokerService.executor().schedule(this::checkReplication, POLICY_UPDATE_FAILURE_RETRY_TIME_SECONDS,
```

Comment:
we are getting exception while executing ```checkReplication``` and in that case we are retrying ```checkReplication``` again. So, if it again fails then it will be an iterative process and it will keep retrying.